newIDE/app/src/ObjectsList/index.js

```diff
@@ -312,7 +312,7 @@ export default class ObjectsListContainer extends React.Component<
     });
   };
 
-  _rename = (objectWithContext: ObjectWithContext, newName: string) => {
+  _rename = (objectWithContext: ObjectWithContext, newName: string) => { ///deprecate for the refactored function
```

Comment:
Can you pass `_canObjectUseNewName` from the parent and use it here :)
Otherwise there will be two place in the code where the logic for checking if an object can be renamed is, and one could forget to update one after working on the other.

Reply:
You can then also removed the comment :)

Reply:
done :)